### PR TITLE
RDoc-699 continue security docs

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown [hunk 1]
```diff
@@ -128,10 +128,13 @@ This options allows you to export the server certificate as a .pfx file. In case
 
 As mentioned above, RavenDB generates client certificates by signing them using the server certificate. A typical server certificate doesn't allow acting as an Intermediate Certificate Authority signing other certificates. This is the case with Let's Encrypt certificates.
 
-In that case, the generated client certificate will have a broken chain of trust, just like in the following screenshot:
+The left screenshot shows a newly generated client certificate, signed by a Let's Encrypt server certificate. You cannot see the full chain of trust because the OS (Windows) doesn't have knowledge of the server certificate.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown]: Left screenshot? Do we control the screenshot placement somehow?

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown [hunk 2]
```diff
@@ -128,10 +128,13 @@ This options allows you to export the server certificate as a .pfx file. In case
...
+
+If you wish to view the full chain, add the server certificate to the OS trusted store. This step is **not necessary** for RavenDB and is explained here only to show how to view the full chain in Windows. The screenshot on the right side shows the full chain.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown]: Right screenshot?

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown [hunk 3]
```diff
@@ -128,10 +128,13 @@ This options allows you to export the server certificate as a .pfx file. In case
...
+The left screenshot shows a newly generated client certificate, signed by a Let's Encrypt server certificate. You cannot see the full chain of trust because the OS (Windows) doesn't have knowledge of the server certificate.
+
+If you wish to view the full chain, add the server certificate to the OS trusted store. This step is **not necessary** for RavenDB and is explained here only to show how to view the full chain in Windows. The screenshot on the right side shows the full chain. 
+
 
 ![Figure 6. Client Cert Chain](images/client-cert.png)
 
-Because client certificates are managed by RavenDB directly and not through any PKI infrastructure <strong>this is perfectly acceptable</strong>. Authenticating a client certificate is done explicitly by looking for the thumbprint in the registered certificates list in the server.
+Because client certificates are managed by RavenDB directly and not through any PKI infrastructure <strong>this is perfectly acceptable</strong>. Authenticating a client certificate is done explicitly by looking for the thumbprint in the registered certificates list in the server and not by validating the chain of trust.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown]: For future: Avoid using raw HTML markup if there is a markdown equivalent,

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/client-certificate-usage.markdown [hunk 4]
```diff
@@ -59,11 +59,24 @@ $cert = Get-PfxCertificate -FilePath c:/secrets/server.pfx
...
 {CODE-BLOCK:powershell}
-wget -UseBasicParsing -Method POST -Certificate $cert -OutFile "cluster.admin.cert.pfx" -Body '{"Name": "cluster.admin.client.certificate","SecurityClearance": "ClusterAdmin","Password": "p@$$w0rd"}' "https://rvn-srv-1:8080/admin/certificates"
+wget -UseBasicParsing -Method POST -Certificate $cert -OutFile "cluster.admin.cert.zip" -Body '{"Name": "cluster.admin.client.certificate","SecurityClearance": "ClusterAdmin","Password": "p@$$w0rd"}' "https://rvn-srv-1:8080/admin/certificates"
 {CODE-BLOCK/}
 
 ### Example III : Using Curl in Linux
 
+At this point you only have a **server certificate** and you will use it (acting as the client certificate).  
+First, we will convert the .pfx certificate to .pem:
+{CODE-BLOCK:plain}
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/client-certificate-usage.markdown]: CODE-BLOCK:bash

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/client-certificate-usage.markdown [hunk 5]
```diff
@@ -59,11 +59,24 @@ $cert = Get-PfxCertificate -FilePath c:/secrets/server.pfx
...
+openssl pkcs12 -in cluster.server.certificate.example.pfx -out server.pem -clcerts
+{CODE-BLOCK/}
+
+{NOTE You must provide a password when creating the .pem file. Curl will only accept a password protected certificate. /}
+
+Then make the request:
+{CODE-BLOCK:plain}
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/client-certificate-usage.markdown]: CODE-BLOCK: bash

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/client-certificate-usage.markdown [hunk 6]
```diff
@@ -59,11 +59,24 @@ $cert = Get-PfxCertificate -FilePath c:/secrets/server.pfx
...
 
 ### Example III : Using Curl in Linux
 
+At this point you only have a **server certificate** and you will use it (acting as the client certificate).  
+First, we will convert the .pfx certificate to .pem:
+{CODE-BLOCK:plain}
+openssl pkcs12 -in cluster.server.certificate.example.pfx -out server.pem -clcerts
+{CODE-BLOCK/}
+
+{NOTE You must provide a password when creating the .pem file. Curl will only accept a password protected certificate. /}
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/client-certificate-usage.markdown]: typo: cURL

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 7]
```diff
@@ -0,0 +1,137 @@
+# Security : Common Errors & Troubleshooting
+
+In this section, we review some of the common security configuration errors and troublshooting.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: typo: troubleshooting

Also I'm not sure if 'review troubleshooting' is a proper phrase?

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+[Setup Wizard](#setup-wizard)
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: use root paths here

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+It can  happen if the ip is external (behind a firewall, docker). If this is the case, try going back to the previous screen and add the same ip as an external ip.
+Settings file:D:\temp\RavenDB-4.0.0-windows-x64\Server\settings.json.
+IP addresses: 10.0.0.65:4433. 
+---> Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvException: Error -4092 EACCES permission denied
+{CODE-BLOCK/}
+
+This can be caused by two different reasons:
+
+1. Your private IP address is not reachable inside the machine or you provided the wrong IP/port combination.
+2. You are running behind a firewall (VM, docker...) and accidentaly provided the external IP address during setup.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: typo: accidentally

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+In this section, we review some of the common security configuration errors and troublshooting.
+
+[Setup Wizard](#setup-wizard)  
+[Authentication](#authentication)  
+[Authorization](#authorization)  
+[Encryption](#encryption)  
+
+## Setup Wizard    
+
+#### Server cannot bind to the provided private IP address
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: h3

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+This can be caused by two different reasons:
+
+1. Your private IP address is not reachable inside the machine or you provided the wrong IP/port combination.
+2. You are running behind a firewall (VM, docker...) and accidentaly provided the external IP address during setup.
+
+Make sure you provide the private IP address in the "IP Address / Hostname" field as seen in [this example](../../start/installation/setup-wizard#example-iii).
+
+
+#### Ports are blocked by the firewall
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: h3

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
System.InvalidOperationException: Setting up RavenDB in Let's Encrypt security mode failed. ---> 
System.InvalidOperationException: Validation failed. ---> 
System.InvalidOperationException: Failed to simulate running the server with the supplied settings using: https://a.example.development.run:443 ---> 
System.InvalidOperationException: Client failed to contact webhost listening to 'https://a.example.development.run:443'.
Are you blocked by a firewall? Make sure the port is open.
Settings file: D:\RavenDB-4.0.0-windows-x64\Server\settings.json.
IP addresses: 10.0.1.4:443.
+{CODE-BLOCK/}
+
+#### DNS is cached locally
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: h3

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+Expected to get these ips: 127.0.0.1 while the actual result was: 10.0.0.65
+If we try resolving through google's api (https://dns.google.com), it works well.
+Try to clear your local/network DNS cache or wait a few minutes and try again.
+Another temporary solution is to configure your local network connection to use google's DNS server (8.8.8.8).
+{CODE-BLOCK/}
+
+This error probably means that the DNS is cached. You can wait a while or reset the network DNS cache but in many cases the easiest solution is to [temporarily switch your DNS server to 8.8.8.8](https://developers.google.com/speed/public-dns/docs/using) 
+You can click the Try Again button to restart the validation process of the Setup Wizard.
+
+#### Long DNS propogation time
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: h3

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+Expected to get these ips: 127.0.0.1 while google's actual result was: 10.0.0.65 
+Please wait a while until DNS propagation is finished and try again. If you are trying to update existing DNS records, 
+it might take hours to update because of DNS caching. If the issue persists, contact RavenDB's support.
+{CODE-BLOCK/}  
+
+If this happens, there is nothing you can do except wait for DNS propogation. When it's updated in dns.google.com click the `Try Again` button.
+
+## Authentication  
+
+#### Getting the full error using powershell
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: h3

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+        $stream = $_.Exception.Response.GetResponseStream()
+        $reader = New-Object System.IO.StreamReader($stream)
+        Write-Host $reader.ReadToEnd()
+    }
+    Write-Error $_.Exception
+}
+{CODE-BLOCK/}
+
+
+#### Not using Tls1.2
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: TLS

Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: h3

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+
+The RavenDB clients use Tls1.2 by default. If you want to use other clients please make sure to use the Tls1.2 security protocol.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: TLS

File: Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/database-encryption.markdown [hunk 17]
```diff
@@ -1,40 +1,79 @@
...
+You must use a client certificate to make the request, because the server is using authentication.
+
+## Windows Example
+
+Load the client certificate in Powershell:
+{CODE-BLOCK:powershell}
+$cert = Get-PfxCertificate -FilePath C:\secrets\admin.client.certificate.example.pfx
+{CODE-BLOCK/}
+
+Make sure to use Tls1.2:
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/database-encryption.markdown]: TLS

File: Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/database-encryption.markdown [hunk 18]
```diff
@@ -1,40 +1,79 @@
...
 
+Finally, create the encrypted database using the Client API:
 {CODE-TABS}
 {CODE-TAB:csharp:Sync CreateEncryptedDatabase@ClientApi\Operations\Server\CreateDeleteDatabase.cs /}
 {CODE-TAB:csharp:Async CreateEncryptedDatabaseAsync@ClientApi\Operations\Server\CreateDeleteDatabase.cs /}
 {CODE-TABS/}
+
+## Remarks
+
+Database encryption must be enabled during creating of the database. If you wish to use encryption in an existing database, it must be exported and then imported back into a new encrypted database.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/database-encryption.markdown]: during creation of

File: Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/encryption-at-rest.markdown [hunk 19]
```diff
@@ -1,39 +1,38 @@
...
 
 
-**What about Encryption in Transit?**
+## What about Encryption in Transit?
 
-To enable encryption in RavenDB, the user must first enable authentication and HTTPS (by providing a certificate).
+To enable encryption in RavenDB, the user must first [enable authentication](../../../server/security/authentication/certificate-configuration) and HTTPS (by providing a certificate).
 
-HTTPS provides privacy and integrity of the data in transit. It protects against man-in-the-middle attacks, eavesdropping and tampering of the communication.
+HTTPS (using the Tls 1.2 protocol) provides privacy and integrity of the data in transit. It protects against man-in-the-middle attacks, eavesdropping and tampering of the communication.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/encryption-at-rest.markdown]: TLS

File: Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/secret-key-management.markdown [hunk 20]
```diff
@@ -8,21 +8,21 @@ In RavenDB this can be handled in two ways - depending on the user's choice:
...
 
 You can provide a master key by setting `Security.MasterKey.Exec` and `Security.MasterKey.Exec.Arguments` in `settings.json`. RavenDB will invoke a process you specify, so you can write your own scripts / mini programs and apply whatever logic you need. It creates a clean separation between RavenDB and the secret store in use. This option is useful when you want to protect your master key with other solutions such as "Azure Key Vault", "HashiCorp Vault" or even Hardware-Based Protection.
 
 RavenDB expects to get a cryptographically secure 256-bit key through the standard output.
 
-For example, the following c# console app (GiveMeKey.cs) will generate a random key and write it to the standard output. Obviously this is just an example, and your executable should supply the same key every time it is invoked.
+For example, the following C# Console Application (GiveMeKey.cs) will generate a random key and write it to the standard output. Obviously this is just an example, and your executable should supply the same key every time it is invoked.
 
 {CODE-BLOCK:csharp}
 using System;
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/secret-key-management.markdown]: can put this into a file?

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown (hunk 3 above)
Comment (iftahbe) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown]: This is old...

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown (hunk 1 above)
Comment (iftahbe) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown]: Look at the image, it contains two screenshots. Left and right.

File: Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/secret-key-management.markdown (hunk 20 above)
Comment (iftahbe) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/secret-key-management.markdown]: I can but then why did you introduce {CODE-BLOCK:csharp} ? 
I actually like it, that I don't have to seek the code in other files.

File: Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown (hunk 7 above)
Comment (iftahbe) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/common-errors-and-troubleshooting.markdown]: yes it's a mistake

File: Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown (hunk 1 above)
Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/authentication/certificate-management.markdown]: Left side maybe?

File: Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/secret-key-management.markdown (hunk 20 above)
Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/server/security/encryption/secret-key-management.markdown]: 1. I haven't introduced it, it was used in docs 1.0 and 2.0 mostly.
2. The benefit of using .cs files is that you are absolutely sure that the code there compiles.

**Pull request merged**